### PR TITLE
CS-1859-CR Datamodel Update and Unit Tests

### PR DESCRIPTION
**Pull request by CyprienRusu**
Ticket: https://pzflex.atlassian.net/browse/CS-1859

I have updated the data model and solved a few issues with the client.
Only 2 functions had different names:
CreateProjectResponse -> CreateProjectRequest
DesignProjectResponse -> DesignProjectRequest
Also there is 1 new function too.

Now, of course, some parameters for each existing functions have also changed, so It needs to be tested throughly.
I have added a tests folder and started to add some unit tests in it

Now, the problem I have is that it works locally because I have installed simAPI and configured the client locally, so I can run pytest, but I am not sure how to handle that in the Github Action script.

Also I think that the client needs simAPI to work, not sure it is installed in the current github action script env

### CODE REVIEW

File: onscale_client/api/rest_api.py [hunk 1]
```diff
@@ -1406,7 +1406,7 @@ def project_create(
         project_title: str = None,
         project_goal: str = None,
         core_hour_limit: int = None,
-    ) -> datamodel.ProjectCreateResponse:
+    ) -> datamodel.ProjectCreateRequest:
```
Comment (jeremyonscale) [inline onscale_client/api/rest_api.py:1409]: I don't understand. How can the method `project_create()` return a request instead of a response?

Comment (jeremyonscale) [reply onscale_client/api/rest_api.py:1409]: The request should remain as a request. I think the data model is missing the ProjectCreateResponse, which should be

```json
{
  "traceId": "string",
  "projectId": "string",
  "accountId": "string",
  "hpcId": "string",
  "userId": "string",
  "projectTitle": "string",
  "projectGoal": "string",
  "createDate": 1585700000000,
  "lastUpdate": 1585700000000,
  "coreHourUsed": 0,
  "designList": [
    {
      "traceId": "string",
      "designId": "design-id",
      "projectId": "string",
      "userId": "string",
      "designTitle": "string",
      "designDescription": "string",
      "designGoal": "string",
      "physics": "MECHANICAL",
      "createDate": 1585700000000,
      "parentDesignId": "string",
      "designInstanceList": [
        {
          "traceId": "string",
          "designInstanceId": "design-instance-id",
          "designId": "design-id",
          "projectId": "string",
          "userId": "string",
          "designInstanceTitle": "Instance Title",
          "description": "string",
          "createDate": 1585700000000,
          "designInstanceHash": "string",
          "parentDesignInstanceId": "string",
          "analysisList": [
            {
              "traceId": "string",
              "analysisId": "analysis-id",
              "projectId": "string",
              "designId": "string",
              "designInstanceId": "string",
              "blobId": "string",
              "accountId": "string",
              "userId": "string",
              "jobId": "string",
              "analysisType": "string",
              "solver": "string",
              "timeDuration": 0,
              "timeStep": 0,
              "createDate": 1585700000000,
              "cipherSalt": "00000000-0000-0000-0000-000000000000",
              "estimateHash": "string",
              "precision": "single",
              "pythonHash": "string",
              "lifeTimeLimit": 0,
              "requiredBlobs": [
                "string"
              ],
              "physicsTypes": [
                "string"
              ]
            }
          ],
          "archived": true,
          "locked": true
        }
      ],
      "archived": true
    }
  ],
  "userIdList": [
    "string"
  ],
  "lastUpdateByMe": 1585700000000,
  "myAccessType": "string",
  "archived": true
}
```

Comment (jeremyonscale) [reply onscale_client/api/rest_api.py:1409]: maybe this is what we should be using? `class Project(BaseModel)`?

Comment (CyprienRusu) [reply onscale_client/api/rest_api.py]: Thanks @jeremyonscale 
You were right, I corrected and changed ProjectCreateRequest into Project and then I implemented some function to list the projects in my account to check if it was working and it does!
The changes are in the last commit: https://github.com/OnScale/OnScale-Cloud-Client/pull/13/commits/374c1b0bc2028c7ab4d514b5edc92c2911a0fd50

I think that all the API calls are available in Swagger, but those for the **Project** model were never really implemented.
This **ProjectCreateResponse** was probably a new model is the process of being implemented, but it never was, so it stayed like that.

File: onscale_client/api/rest_api.py [hunk 2]
```diff
@@ -1454,14 +1454,54 @@ def project_create(
...
+
+        Examples:
+            >>> import onscale_client.api.rest_api as rest_api
+            >>> api = rest_api.RestApi(portal='prod', auth_token='AUTH_TOKEN')
+            >>> proj_list = api.project_list(account_id=0954e70b-237a-4cdb-a267-b5da0f67dd70)
+            >>> print(proj_list[0].project_title)
+            'My OnScale Project'
+        """
+        if self.debug_output:
+            print("RestApi.account_list:")
```
Comment (jeremyonscale) [inline onscale_client/api/rest_api.py:1486]: project instead of account

File: onscale_client/api/rest_api.py (hunk 1 above)
Comment (jeremyonscale) [reply onscale_client/api/rest_api.py]: Looks good. Let me try it out.

File: onscale_client/api/rest_api.py (hunk 2 above)
Comment (CyprienRusu) [reply onscale_client/api/rest_api.py]: Ok, corrected in https://github.com/OnScale/OnScale-Cloud-Client/pull/13/commits/4028e2c27b36fbb7850d40f137f0a56deada89e6

**Review by smcclennan — APPROVED** [review]

**Pull request merged**